extension/js/add_input_icon.js

```diff
@@ -88,11 +90,37 @@ function addRelayIconToInput(emailInput) {
   emailInput.remove();
 }
 
-(function () {
+function getEmailInputsAndAddIcon() {
   const getEmailInputs = document.querySelectorAll("input[type='email']");
   for (const emailInput of getEmailInputs) {
-    if (!emailInput.querySelector(".relay-icon")) {
+    if (!emailInput.parentElement.classList.contains("relay-email-input-wrapper")) {
       addRelayIconToInput(emailInput);
     }
   }
+}
+
+(function() {
+  // Catch all immediately findable email inputs
+  getEmailInputsAndAddIcon();
+
+  // Catch email inputs that only become findable after
+  // the entire page (including JS/CSS/images/etc) is fully loaded.
+  window.addEventListener("load", () => {
+    getEmailInputsAndAddIcon();
+  });
+
+  // Create MutationObserver and watch for dynamically generated email inputs
+  const mutationObserver = new MutationObserver(function(mutations) {
+    mutations.forEach(function(mutation) {
+      if (mutation.target.tagName === "FORM") {
+        const emailInput = mutation.target.querySelector("input[type='email']");
+        if (emailInput) {
+          addRelayIconToInput(emailInput);
+        }
+      }
+    });
+  });
+
+  mutationObserver.observe(document.body, { childList: true, subtree: true });
```

Comment:
We should do a performance check on this to see if we can optimize this. There are some tips in [this answer](https://stackoverflow.com/a/39332340/13373794), but I don't know if any of them apply to our use-case?

Doesn't need to block merge, in any case.